From 7b028bd667664bf2bc5c681929f90c2c5fbf7c44 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 28 Apr 2015 21:43:17 +0100
Subject: [PATCH 07/10] add many files to installed headers

tools/projectgeneratormanager.h
tools/id.h
language/filetags.h
logging/logger.h
generators/generator.h
---
 src/lib/corelib/generators/generators.pri | 6 +++++
 src/lib/corelib/language/language.pri     | 3 ++-
 src/lib/corelib/logging/logging.pri       | 3 ++-
 src/lib/corelib/tools/tools.pri           | 2 ++
 4 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/src/lib/corelib/generators/generators.pri b/src/lib/corelib/generators/generators.pri
index 2fa095e..4217853 100644
--- a/src/lib/corelib/generators/generators.pri
+++ b/src/lib/corelib/generators/generators.pri
@@ -8,6 +8,12 @@
     $$PWD/generator.h \
     $$PWD/generatordata.h \
     $$PWD/igeneratableprojectvisitor.h
+
+!qbs_no_dev_install {
+    generators_headers.files = $$PWD/generator.h
+    generators_headers.path = $${QBS_INSTALL_PREFIX}/include/qbs/generators
+    INSTALLS += generators_headers
+}
 
 include(clangcompilationdb/clangcompilationdb.pri)
 include(visualstudio/visualstudio.pri)
diff --git a/src/lib/corelib/language/language.pri b/src/lib/corelib/language/language.pri
index 6da8536..4f8f2ba 100644
--- a/src/lib/corelib/language/language.pri
+++ b/src/lib/corelib/language/language.pri
@@ -67,7 +67,8 @@ qbs_enable_unit_tests {
 }
 
 !qbs_no_dev_install {
-    language_headers.files = $$PWD/forward_decls.h
+    language_headers.files = $$PWD/forward_decls.h \
+                             $$PWD/filetags.h
     language_headers.path = $${QBS_INSTALL_PREFIX}/include/qbs/language
     INSTALLS += language_headers
 }
diff --git a/src/lib/corelib/logging/logging.pri b/src/lib/corelib/logging/logging.pri
index 8c03677..c09ba11 100644
--- a/src/lib/corelib/logging/logging.pri
+++ b/src/lib/corelib/logging/logging.pri
@@ -8,7 +8,8 @@ SOURCES += \
     $$PWD/ilogsink.cpp
 
 !qbs_no_dev_install {
-    logging_headers.files = $$PWD/ilogsink.h
+    logging_headers.files = $$PWD/ilogsink.h \
+                            $$PWD/logger.h
     logging_headers.path = $${QBS_INSTALL_PREFIX}/include/qbs/logging
     INSTALLS += logging_headers
 }
diff --git a/src/lib/corelib/tools/tools.pri b/src/lib/corelib/tools/tools.pri
index d5f5983..cc20c33 100644
--- a/src/lib/corelib/tools/tools.pri
+++ b/src/lib/corelib/tools/tools.pri
@@ -95,6 +95,8 @@ qbs_enable_unit_tests {
         $$PWD/processresult.h \
         $$PWD/qbs_export.h \
         $$PWD/buildoptions.h \
+        $$PWD/projectgeneratormanager.h \
+        $$PWD/id.h \
         $$PWD/generateoptions.h \
         $$PWD/generatorpluginmanager.h \
         $$PWD/installoptions.h \
-- 
2.3.5

